googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -1291,16 +1291,49 @@ constexpr bool InstantiateTypedTestCase_P_IsDeprecated() { return true; }
   } else                     /* NOLINT */                         \
     static_assert(true, "")  // User must have a semicolon after expansion.
 
+#if GTEST_HAS_EXCEPTIONS
+
+#if GTEST_HAS_RTTI
+
+#define GTEST_EXCEPTION_TYPE_(e) \
+  ::testing::internal::GetTypeName(typeid(e))
+
+#else  // GTEST_HAS_RTTI
+
+#define GTEST_EXCEPTION_TYPE_(e) \
+  std::string{"an std::exception-derived error"}
+
+#endif  // GTEST_HAS_RTTI
+
+#define GTEST_TEST_THROW_CATCH_STD_EXCEPTION_(statement, expected_exception) \
+  catch (std::exception const& e) { \
```

Comment:
This will unfortunately fail if the user'ss expected_exception is, itself, std::exception, since we will have two catch clauses with the same type. One way we might avoid this is do a tiny bit of metaprogramming here and basically have it catch something like

```
catch(
    typename std::conditional<
        std::is_same<
            typename std::remove_cv<
                typename std::remove_reference<expected_exception>::type>::type,
            std::exception>,
            const SomeInternalTypeWithAWhatFunction&,
            const std::exception&
        >::type e)
```

Reply:
There may be alternative solutions here, this is just the first solution I can think of.

Reply:
If I understand the standard correctly, it shouldn't. `catch` clauses are matched from top to bottom, and the first one that matches the exception is called. If we have two `catch`es with the same type, the later one will be ignored.

From cppreference:

> When an exception of type E is thrown by any statement in compound-statement, it is matched against the types of the formal parameters T of each catch-clause in handler-seq, in the order in which the catch clauses are listed. The exception is a match if any of the following is true: ...

Reply:
True, my mistake -- it's not an error, but clang at least does warn on duplicate catches by default:

https://gcc.godbolt.org/z/DC5KSS

Regardless, I think we should do what we can to avoid such warnings here.

Reply:
I've fixed the issue and added a test for this case.